kvbc/src/ReplicaImp.cpp

```diff
@@ -277,39 +259,20 @@ void ReplicaImp::insertBlockInternal(BlockId blockId, Sliver block) {
                 "Block size test " << (existingBlock.length() != block.length()) << ", block data test "
                                    << (memcmp(existingBlock.data(), block.data(), block.length())));
 
-      m_bcDbAdapter->deleteBlockAndItsKeys(blockId);
+      m_bcDbAdapter->deleteBlock(blockId);
 
       // TODO(GG): how do we want to handle this - restart replica?
       // exit(1);
       return;
     }
-  } else {
-    s = m_bcDbAdapter->addBlock(block, blockId);
-    if (!s.isOK()) {
-      // TODO(SG): What to do?
-      printf("Failed to add block");
-      exit(1);
-    }
+  } catch (const NotFoundException &e) {
+    m_bcDbAdapter->addRawBlock(block, blockId);  // TODO [TK] will be replaced after integration of hasBlock
```

Comment:
Is the expectation that this throws on error and is purposefully uncaught?

Reply:
Yes, previously there were exit(1) directly in places they were thrown, and now the exceptions are caught at the most outer layer